src/components/ha-menu-button.ts

```diff
@@ -5,34 +5,113 @@ import {
   LitElement,
   html,
   customElement,
+  CSSResult,
+  css,
 } from "lit-element";
 
 import { fireEvent } from "../common/dom/fire_event";
+import { HomeAssistant } from "../types";
+import { UnsubscribeFunc } from "home-assistant-js-websocket";
+import { subscribeNotifications } from "../data/persistent_notification";
+import computeDomain from "../common/entity/compute_domain";
 
 @customElement("ha-menu-button")
 class HaMenuButton extends LitElement {
-  @property({ type: Boolean })
-  public hassio = false;
+  @property({ type: Boolean }) public hassio = false;
+  @property() public narrow!: boolean;
+  @property() public hass!: HomeAssistant;
+  @property() private _hasNotifications = false;
+  private _attachNotifOnConnect = false;
+  private _unsubNotifications?: UnsubscribeFunc;
+
+  public connectedCallback() {
+    super.connectedCallback();
+    if (this._attachNotifOnConnect) {
+      this._attachNotifOnConnect = false;
+      this._subscribeNotifications();
+    }
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    if (this._unsubNotifications) {
+      this._attachNotifOnConnect = true;
+      this._unsubNotifications();
+      this._unsubNotifications = undefined;
+    }
+  }
 
   protected render(): TemplateResult | void {
+    const hasNotifications =
+      this.narrow &&
+      (this._hasNotifications ||
+        Object.keys(this.hass.states).some(
+          (entityId) => computeDomain(entityId) === "configurator"
+        ));
     return html`
       <paper-icon-button
         aria-label="Sidebar Toggle"
         .icon=${this.hassio ? "hassio:menu" : "hass:menu"}
         @click=${this._toggleMenu}
       ></paper-icon-button>
+      ${hasNotifications
+        ? html`
+            <div class="dot"></div>
+          `
+        : ""}
     `;
   }
 
-  // We are not going to use ShadowDOM as we're rendering a single element
-  // without any CSS used.
-  protected createRenderRoot(): Element | ShadowRoot {
-    return this;
+  protected updated(changedProps) {
+    super.updated(changedProps);
+
+    if (!changedProps.has("narrow")) {
+      return;
+    }
+
+    this.style.visibility = this.narrow ? "initial" : "hidden";
```

Comment:
Wouldn't `display: none` be better? Now you have a lot of margin in front of the header text when the button is hidden.

Reply:
It would misalign the text on Lovelace, I don't know what I dislike more.

Reply:
With the views menu?

Reply:
![60402023-81d4df00-9b3e-11e9-8d1b-e1b06249dd12](https://user-images.githubusercontent.com/1444314/60452832-56a4cb00-9be4-11e9-9fbe-be683c95e9a8.png)

Reply:
What about adjusting the margin of both app-toolbar & paper-tabs on narrow?

Reply:
Which margin?